### ISSUE TITLE
Floats incorrectly serialized to xml

### ISSUE THREAD

**Issue by peterlozano** [schmittjoh/serializer#1284]
After an upgrade from 1.14.1 to 3.11.0, float values were not serialized correctly.

For example:
348.47 was serialized to xml as 348.46999999999997

I'm using PHP 7.2.23

It seems var_export may format floats this way for some reason: https://github.com/schmittjoh/serializer/blob/f91971c8e5de8d8c9ab11a338f9c1d5189a13762/src/XmlSerializationVisitor.php#L182

See:
* https://stackoverflow.com/questions/32149340/php-var-export-fails-with-float

**Comment by zevnikrok** [comment]
I am experiencing the same issue after upgrading to v3. 
Is there any workaround or plan to change the behaviour?

**Comment by goetas** [comment]
an option would be to introduce a "precision" parameter to float numbers.... something linke `float<1>` could serialize `348.47` into `348.5`

**Comment by machinateur** [comment]
Hello everyone,

I've experienced the same problem earlier today and could not find any information on how to format `float` values in the docs or elsewhere. I ended up using `round()` to fix it (hopefully, I haven't been able to test it properly).

Also, I'm unable to reproduce the results described by @peterlozano on [3v4l](https://3v4l.org/eon9q) cutting the middleman and directly giving the float to `var_export()`. So I can't even reproduce the behavior I was getting myself... Am I missing something?

Best regards
